src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServerCapability.cs

```diff
@@ -1,39 +1,35 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer
 {
-    internal record RazorLanguageServerCapability(
-        bool LanguageQuery,
-        bool RangeMapping,
-        bool EditMapping,
-        bool MonitorProjectConfigurationFilePath,
-        bool BreakpointSpan,
-        bool ProximityExpressions)
+    internal class RazorLanguageServerCapability : IRegistrationExtension
     {
         private const string RazorCapabilityKey = "razor";
-        private static readonly RazorLanguageServerCapability s_default = new RazorLanguageServerCapability(
-            LanguageQuery: true,
-            RangeMapping: true,
-            EditMapping: true,
-            MonitorProjectConfigurationFilePath: true,
-            BreakpointSpan: true,
-            ProximityExpressions: true);
+        private static readonly RazorLanguageServerCapability s_default = new RazorLanguageServerCapability
+        {
+            LanguageQuery = true,
+            RangeMapping = true,
+            EditMapping = true,
+            MonitorProjectConfigurationFilePath = true,
+            BreakpointSpan = true,
+            ProximityExpressions = true
+        };
+
+        public bool LanguageQuery { get; set; }
+        public bool RangeMapping { get; set; }
+        public bool EditMapping { get; set; }
+        public bool MonitorProjectConfigurationFilePath { get; set; }
+        public bool BreakpointSpan { get; set; }
+        public bool ProximityExpressions { get; set; }
 
-        public static void AddTo(ServerCapabilities capabilities)
+        public RegistrationExtensionResult GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
-            // We have to use the experimental capabilities bucket here because not all platforms maintain custom capabilities. For instance
-            // in Visual Studio scenarios it will deserialize server capabilties into what it believes is "valid" and then will re-pass said
-            // server capabilities to our client side code having lost the information of the custom capabilities. To avoid this we use the
-            // experimental bag since it's part of the official LSP spec. This approach enables us to work with any client.
-            capabilities.Experimental ??= new Dictionary<string, JToken> {
-                { RazorCapabilityKey, JToken.FromObject(s_default) }
-            };
+            return new RegistrationExtensionResult(RazorCapabilityKey, JToken.FromObject(s_default));
```

Comment:
This still ends up using experimental capabilties, thanks to the code in `IServiceCollectionExtensions`, just without the scary comment.